acceptance/analyzer_test.go

```diff
@@ -0,0 +1,599 @@
+					// Resolve it so it can be properly mounted by the Docker daemon.
+					dockerConfig, err = filepath.EvalSymlinks(dockerConfig)
+					h.AssertNil(t, err)
+
+					_, tempDir := h.DockerRunAndCopy(t,
+						analyzeImage,
+						"/layers",
+						h.WithFlags(
+							"--mount", fmt.Sprintf("type=bind,source=%s,target=/mounted-docker-config", dockerConfig),
+							"--env", "DOCKER_CONFIG=/mounted-docker-config",
+							"--network", "host",
+						),
+						h.WithBash(
+							fmt.Sprintf("chown -R 2222:3333 /mounted-docker-config; %s %s; ls -alR /layers", analyzerPath, appImage), // provide a real app image, so that we can test that the registry is accessible
+						),
+					)
+					defer os.RemoveAll(tempDir)
+
+					h.AssertPathExists(t, filepath.Join(tempDir, "layers", "some-buildpack-id"))
+				})
+			})
+		})
+
+		when("cache is provided", func() {
+			when("cache image case", func() {
+				var cacheImage, cacheAuthConfig string
+
+				it.Before(func() {
+					metadata := flattenMetadata(t, filepath.Join("testdata", "analyzer", "cache_image_metadata.json"), lifecycle.CacheMetadata{})
+					cacheImage, cacheAuthConfig = buildRegistryImage(
+						t,
+						"some-cache-image-"+h.RandString(10),
+						filepath.Join("testdata", "analyzer", "cache-image"),
+						"--build-arg", "fromImage="+variables.ContainerBaseImage,
+						"--build-arg", "metadata="+metadata,
+					)
+				})
+
+				it.After(func() {
+					h.DockerImageRemove(t, cacheImage)
+				})
+
+				it("uses the provided cache", func() {
+					_, tempDir := h.DockerRunAndCopy(t,
+						analyzeImage,
+						"/layers",
+						h.WithFlags(
+							"--network", "host",
+							"--env", "CNB_REGISTRY_AUTH="+cacheAuthConfig,
+						),
+						h.WithArgs(
+							analyzerPath,
+							"-cache-image", cacheImage,
+							"some-image",
+						),
+					)
+					defer os.RemoveAll(tempDir)
+
+					h.AssertPathExists(t, filepath.Join(tempDir, "layers", "some-buildpack-id", "some-layer.sha"))
+					h.AssertPathExists(t, filepath.Join(tempDir, "layers", "some-buildpack-id", "some-layer.toml"))
+				})
+			})
+
+			when("cache directory case", func() {
+				it("uses the provided cache", func() {
+					cacheVolume := h.SeedDockerVolume(t, cacheFixtureDir)
+					defer h.DockerVolumeRemove(t, cacheVolume)
+
+					_, tempDir := h.DockerRunAndCopy(t,
+						analyzeImage,
+						"/layers",
+						h.WithFlags(
+							"--env", "CNB_REGISTRY_AUTH={}",
+							"--volume", cacheVolume+":"+"/cache",
+						),
+						h.WithArgs(
+							analyzerPath,
+							"-cache-dir", "/cache",
+							"some-image",
+						),
+					)
+					defer os.RemoveAll(tempDir)
+
+					h.AssertPathExists(t, filepath.Join(tempDir, "layers", "some-buildpack-id", "some-layer.sha"))
+					h.AssertPathExists(t, filepath.Join(tempDir, "layers", "some-buildpack-id", "some-layer.toml"))
+				})
+			})
+		})
+	})
+}
+
+func flattenMetadata(t *testing.T, path string, metadataStruct interface{}) string {
```

Comment:
nit: maybe we can call this `minifyMetadata` instead of `flattenMetadata`? To me flatten implies something different (flattening nested arrays).